src/toast.tsx

```diff
@@ -1,366 +1,368 @@
-    Animated.timing(getPanResponderAnim(), {
-      toValue: { x: (dims.width / 10) * 9, y: gestureState.dy },
-      useNativeDriver: Platform.OS !== "web",
-      duration: 250,
-    }).start(() => onDestroy());
-  };
-
-  const getPanResponder = () => {
-    if (panResponderRef.current) return panResponderRef.current;
-    panResponderRef.current = PanResponder.create({
-      onMoveShouldSetPanResponder: (_, gestureState) => {
-        //return true if user is swiping, return false if it's a single click
-        return !(gestureState.dx === 0 && gestureState.dy === 0);
-      },
-      onPanResponderMove: (_, gestureState) => {
-        getPanResponderAnim()?.setValue({
-          x: gestureState.dx,
-          y: gestureState.dy,
-        });
-      },
-      onPanResponderRelease: (_, gestureState) => {
-        if (gestureState.dx > 50) {
-          panReleaseToRight(gestureState);
-        } else if (gestureState.dx < -50) {
-          panReleaseToLeft(gestureState);
-        } else {
-          Animated.spring(getPanResponderAnim(), {
-            toValue: { x: 0, y: 0 },
+    const panReleaseToRight = (gestureState: PanResponderGestureState) => {
+        Animated.timing(getPanResponderAnim(), {
+            toValue: { x: (dims.width / 10) * 9, y: gestureState.dy },
             useNativeDriver: Platform.OS !== "web",
-          }).start();
+            duration: 250,
+        }).start(() => onDestroy());
+    };
+
+    const getPanResponder = () => {
+        if (panResponderRef.current) return panResponderRef.current;
+        panResponderRef.current = PanResponder.create({
+            onMoveShouldSetPanResponder: (_, gestureState) => {
+                //return true if user is swiping, return false if it's a single click
+                return !(gestureState.dx === 0 && gestureState.dy === 0);
+            },
+            onPanResponderMove: (_, gestureState) => {
+                getPanResponderAnim()?.setValue({
+                    x: gestureState.dx,
+                    y: gestureState.dy,
+                });
+            },
+            onPanResponderRelease: (_, gestureState) => {
+                if (gestureState.dx > 50) {
+                    panReleaseToRight(gestureState);
+                } else if (gestureState.dx < -50) {
+                    panReleaseToLeft(gestureState);
+                } else {
+                    Animated.spring(getPanResponderAnim(), {
+                    toValue: { x: 0, y: 0 },
+                    useNativeDriver: Platform.OS !== "web",
+                }).start();
+                }
+            },
+        });
+        return panResponderRef.current;
+    };
+
+    const getPanResponderAnim = () => {
+        if (panResponderAnimRef.current) return panResponderAnimRef.current;
+        panResponderAnimRef.current = new Animated.ValueXY({ x: 0, y: 0 });
+        return panResponderAnimRef.current;
+    };
+
+    if (icon === undefined) {
+        switch (type) {
```

Comment:
i would combine this two switch statements into one, since
they are comparing the same type